internal/bminventory/inventory.go

```diff
@@ -933,7 +933,7 @@ func (b *bareMetalInventory) updateExternalImageInfo(ctx context.Context, infraE
 		}
 
 		details := b.getIgnitionConfigForLogging(ctx, infraEnv, b.log, imageType)
-		eventgen.SendImageInfoUpdatedEvent(ctx, b.eventsHandler, *infraEnv.ID, details)
+		eventgen.SendImageInfoUpdatedEvent(ctx, b.eventsHandler, &infraEnv.ClusterID, *infraEnv.ID, details)
```

Comment:
Is this field still being set by the UI? Otherwise we'll need to look the cluster up, right?
Also whats the difference between setting a pointer to an empty string vs nil here?

Reply:
Checked and I see that the UI used POST `api/assisted-install/v2/infra-envs/` (A.K.A `RegisterInfraEnv`) and the payload included the `cluster_id`

Reply:
pointer to empty string is not nil, so this event will always send an empty string for cluster id no?

Reply:
Yes. It should send a pointer to ClusterID also because there are cases (not by UI) that the infraenv will not be bound

Reply:
Valid point. Addressing this issue here: https://github.com/openshift/assisted-service/pull/3623